undertow/src/main/java/org/wildfly/extension/undertow/deployment/SharedSessionManagerDeploymentProcessor.java

```diff
@@ -114,7 +115,11 @@ public Duration getDefaultSessionTimeout() {
             }
         };
         provider.getSessionManagerFactoryServiceConfigurator(managerServiceName, configuration).configure(support).build(target).install();
-        provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration).configure(support).build(target).install();
+        if (servletContainer == null || servletContainer.getAffinityCookieConfig() == null) {
+            provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration).configure(support).build(target).install();
+        } else {
+            provider.getAffinityLocatorServiceConfigurator(affinityServiceName, configuration).configure(support).build(target).install();
+        }
```

Comment:
We should always install both a session identifier codec service and an affinity locator service.
Whether a given subdeployment WAR will use the former or the latter will depend on which servlet container it is configured to use.

Reply:
Fixed now.